<h1>Data Analysis - Finding the golden age of video game sales</h1>

<h2>Project Objective</h2>

In this project, I will be analyzing data to determine the "golden age" of video game sales — the time period marked by peak quality and commercial success for the video games industry. Using a dataset that includes 999 games released between 1978-2020, I will examine trends in sales and ratings over time to highlight key years that defined the most significant era in gaming history.

The dataset we'll be working with contains 2 tables with the records shown below. The complete open-source dataset can be found <a href="https://www.kaggle.com/datasets/holmjason2/videogamedata">here</a> on Kaggle, with over 13,000 video games' data.


<b>game_sales_data:-</b>
| column | type | description |
|------|---------|-------|
|name|text|Name of the video game|
|platform|text|Platform|
|publisher|text|Game publisher|
|developer|text|Game developer|
|total_shipped|double|Number of copies sold
|year|int|Release year|

<b>reviews:-</b>
| column | type | description |
|------|---------|-------|
|name|text|Name of the video game|
|critic_score|text|Critic score according to Metacritic|
|user_score|text|User score according to Metacritic|

In [12]:
import pymysql
import pandas as pd

In [13]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [14]:
%sql mysql+mysqldb://root:******@localhost/videogames

In [15]:
%config SqlMagic.style = '_DEPRECATED_DEFAULT'

In [16]:
%%sql

-- Summary of first table, sorted by units sold
SELECT * FROM game_sales_data
ORDER BY Total_Shipped DESC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/videogames
10 rows affected.


[('Wii Sports', 'Wii', 'Nintendo', 'Nintendo EAD', 82.9, 2006),
 ('Super Mario Bros.', 'NES', 'Nintendo', 'Nintendo EAD', 40.24, 1985),
 ('Counter-Strike: Global Offensive', 'PC', 'Valve', 'Valve Corporation', 40.0, 2012),
 ('Mario Kart Wii', 'Wii', 'Nintendo', 'Nintendo EAD', 37.32, 2008),
 ("PLAYERUNKNOWN'S BATTLEGROUNDS", 'PC', 'PUBG Corporation', 'PUBG Corporation', 36.6, 2017),
 ('Minecraft', 'PC', 'Mojang', 'Mojang AB', 33.15, 2010),
 ('Wii Sports Resort', 'Wii', 'Nintendo', 'Nintendo EAD', 33.13, 2009),
 ('Pokemon Red / Green / Blue Version', 'GB', 'Nintendo', 'Game Freak', 31.38, 1998),
 ('New Super Mario Bros.', 'DS', 'Nintendo', 'Nintendo EAD', 30.8, 2006),
 ('New Super Mario Bros. Wii', 'Wii', 'Nintendo', 'Nintendo EAD', 30.3, 2009)]

<h2>Cleaning data - finding 'null' values</h2>

Based on the data above, we can see the best-selling video games were released between 1985-2009. Before we dive deeper, we will look for limitations in our database that could cause discrepancies in our analysis.

In [17]:
%%sql

-- Finding games with NULL reviews
SELECT *
FROM game_sales_data
LEFT JOIN reviews
USING(name)
WHERE Critic_Score IS NULL AND User_Score IS NULL

 * mysql+mysqldb://root:***@localhost/videogames
164 rows affected.


[('Minecraft', 'PC', 'Mojang', 'Mojang AB', 33.15, 2010, None, None),
 ('Minecraft', 'PC', 'Mojang', 'Mojang AB', 33.15, 2010, None, None),
 ('Tetris', 'GB', 'Nintendo', 'Bullet Proof Software', 30.26, 1989, None, None),
 ('Super Mario Land', 'GB', 'Nintendo', 'Nintendo R&D1', 18.14, 1989, None, None),
 ("Garry's Mod", 'PC', 'Valve', 'Facepunch Studios', 16.9, 2006, None, None),
 ('Call of Duty: Black Ops 3', 'PS4', 'Activision', 'Treyarch', 15.09, 2015, None, None),
 ('Minecraft', 'X360', 'Microsoft Studios', 'Mojang', 13.0, 2013, None, None),
 ('Minecraft', 'X360', 'Microsoft Studios', 'Mojang', 13.0, 2013, None, None),
 ('Call of Duty: Black Ops IIII', 'PS4', 'Activision', 'Treyarch', 9.32, 2018, None, None),
 ('FIFA 13', 'PS3', 'EA Sports', 'EA Canada', 8.01, 2012, None, None),
 ('FIFA 13', 'PS3', 'EA Sports', 'EA Canada', 8.01, 2012, None, None),
 ('Call of Duty: Black Ops 3', 'XOne', 'Activision', 'Treyarch', 7.37, 2015, None, None),
 ('Yokai Watch 2: Bony Spirits / Fleshy Souls / Psychic Specters', '3DS', 'Nintendo', 'Level 5', 7.29, 2016, None, None),
 ('Just Dance 4', 'Wii', 'Ubisoft', 'Ubisoft', 6.89, 2012, None, None),
 ('Minecraft', 'PS4', 'Sony Computer Entertainment', 'Mojang', 6.33, 2014, None, None),
 ('Minecraft', 'PS4', 'Sony Computer Entertainment', 'Mojang', 6.33, 2014, None, None),
 ('Street Fighter II: The World Warrior', 'SNES', 'Capcom', 'Capcom', 6.3, 1992, None, None),
 ('Minecraft', 'PS3', 'Sony Computer Entertainment America', 'Mojang', 6.05, 2014, None, None),
 ('Minecraft', 'PS3', 'Sony Computer Entertainment America', 'Mojang', 6.05, 2014, None, None),
 ('Sonic the Hedgehog 2', 'GEN', 'Sega', 'Sonic Team', 6.0, 1992, None, None),
 ('Tetris', 'NES', 'Nintendo', 'Nintendo', 5.58, 1989, None, None),
 ('Minecraft', 'XOne', 'Microsoft Studios', 'Mojang', 5.43, 2014, None, None),
 ('Minecraft', 'XOne', 'Microsoft Studios', 'Mojang', 5.43, 2014, None, None),
 ('Dr. Mario', 'GB', 'Nintendo', 'Nintendo R&D1', 5.34, 1990, None, None),
 ('Dr. Mario', 'GB', 'Nintendo', 'Nintendo R&D1', 5.34, 1990, None, None),
 ('Gran Turismo 6', 'PS3', 'Sony Computer Entertainment America', 'Polyphony Digital', 5.22, 2013, None, None),
 ("Kirby's Dream Land", 'GB', 'Nintendo', 'HAL Laboratory', 5.13, 1992, None, None),
 ('FIFA 13', 'X360', 'EA Sports', 'EA Canada', 5.11, 2012, None, None),
 ('FIFA 13', 'X360', 'EA Sports', 'EA Canada', 5.11, 2012, None, None),
 ('Monster Hunter Freedom 3', 'PSP', 'Capcom', 'Capcom', 4.9, 2010, None, None),
 ('Call of Duty: Black Ops IIII', 'XOne', 'Activision', 'Treyarch', 4.85, 2018, None, None),
 ('Dr. Mario', 'NES', 'Nintendo', 'Nintendo R&D1', 4.85, 1990, None, None),
 ('Dr. Mario', 'NES', 'Nintendo', 'Nintendo R&D1', 4.85, 1990, None, None),
 ('Star Wars Battlefront II (2017)', 'PS4', 'Electronic Arts', 'EA DICE', 4.53, 2017, None, None),
 ('Arma 2: Operation Arrowhead', 'PC', 'Meridian4', 'Bohemia Interactive', 4.51, 2010, None, None),
 ('Riven: The Sequel to Myst', 'PC', 'Red Orb', 'Cyan Worlds', 4.5, 1997, None, None),
 ('Pitfall!', '2600', 'Activision', 'Activision', 4.5, 1982, None, None),
 ("The Legend of Zelda: Link's Awakening", 'NS', 'Nintendo', 'Nintendo', 4.38, 2019, None, None),
 ('Zelda II: The Adventure of Link', 'NES', 'Nintendo', 'Nintendo EAD', 4.38, 1988, None, None),
 ("Tom Clancy's Rainbow Six: Siege", 'PS4', 'Ubisoft', 'Ubisoft Montreal', 4.36, 2015, None, None),
 ("Tom Clancy's Rainbow Six: Siege", 'PS4', 'Ubisoft', 'Ubisoft Montreal', 4.36, 2015, None, None),
 ('7 Days to Die', 'PC', 'The Fun Pimps', 'The Fun Pimps', 4.18, 2013, None, None),
 ('Frogger', 'PS', 'Hasbro Interactive', 'Millenium Interactive', 4.16, 1997, None, None),
 ("Assassin's Creed: Unity", 'PS4', 'Ubisoft', 'Ubisoft Montreal', 4.14, 2014, None, None),
 ("Assassin's Creed: Unity", 'PS4', 'Ubisoft', 'Ubisoft Montreal', 4.14, 2014, None, None),
 ('Monster Hunter 4', '3DS', 'Capcom', 'Capcom', 4.1, 2013, None, None),
 ('Namco Museum Vol.3', 'PS', 'Namco', 'Namco', 4.05, 1997, None, None),
 ('Golf'

<h2> Top years based on critic reviews </h2>

The query above helps us find 164 records with missing review scores.  It is a fairly small percentage of null values but important to keep note of. We will continue with our analysis, finding the years with the top critic scores.

In [23]:
%%sql

-- Finding average critic scores, grouped by release year and sorted by average score
SELECT year, ROUND(AVG(reviews.Critic_Score), 1) AS avg_critic_score
FROM game_sales_data AS sales
LEFT JOIN reviews
USING(name)
GROUP BY sales.year
ORDER BY avg_critic_score DESC
LIMIT 10

 * mysql+mysqldb://root:***@localhost/videogames
10 rows affected.


[(1990, 9.4),
 (1992, 9.2),
 (2020, 9.0),
 (2000, 8.6),
 (1993, 8.6),
 (2014, 8.6),
 (2011, 8.6),
 (2009, 8.6),
 (1994, 8.6),
 (2012, 8.5)]

<h2> Exploring the top critically acclaimed years </h2>

Grouping the games by their release year and finding the average critic review score for each year shows us some of the years loved by critics. The year 1990 is on top with an average critic score of 9.4, but how many games were released in 1990? We will now find the number of games released by year, allowing for a larger sample data to work with.

In [29]:
%%sql

-- Finding average critic score, grouped by release years, where number of games released per year was more than 50
SELECT year, ROUND(AVG(Critic_Score), 2) as avg_critic_score, COUNT(g.name) as num_games
FROM game_sales_data as g
INNER JOIN reviews USING(name)
GROUP BY year
HAVING num_games >= 50
ORDER by avg_critic_score DESC
LIMIT 10

 * mysql+mysqldb://root:***@localhost/videogames
10 rows affected.


[(2011, 8.59, 98),
 (2014, 8.57, 102),
 (2009, 8.57, 83),
 (2013, 8.53, 115),
 (2012, 8.47, 52),
 (2015, 8.45, 79),
 (2016, 8.4, 63),
 (2010, 8.38, 98),
 (2008, 8.1, 91),
 (2007, 8.0, 93)]

<h2> Best years for gaming according to the players </h2>

Now that we have the number of games released per year, we can focus on the years with a more substantial number of releases. Release years with a higher number of games will increase the relevance of our analysis, improving the reliability of our conclusions.

Based on our data above, looks like the early 2010s had some great years for video games based on critic score. Let's run the same query to find the top years based on user reviews.

In [30]:
%%sql

-- Same criteria, except using user score instead of critic score
SELECT year, ROUND(AVG(r.user_score), 2) AS avg_user_score, COUNT(g.Name) AS num_games
FROM game_sales_data AS g 
INNER JOIN reviews AS r USING (Name)
GROUP BY g.year
HAVING num_games > 50
ORDER by avg_user_score DESC
LIMIT 10

 * mysql+mysqldb://root:***@localhost/videogames
10 rows affected.


[(2009, 9.15, 83),
 (2010, 9.09, 98),
 (2008, 8.9, 91),
 (2007, 8.67, 93),
 (2011, 8.18, 98),
 (2012, 7.38, 52),
 (2013, 7.18, 115),
 (2015, 7.12, 79),
 (2014, 6.89, 102),
 (2016, 6.75, 63)]

<h2> Comparing the top years between users and critics </h2>

Results from the players' perspectives seems to tell a similar story, with late 2000's and early 2010's coming out on top. Now that we have the peak years for both users and critics, let's find the common years between the 2 sets, also including the total sales per year to reach a more definite conclusion in our analysis.

In [35]:
%%sql


-- Finding the top years based on the average critic and user score being higher or equal to 8.5, with annual sales

SELECT year, ROUND(AVG(r.user_score), 1) AS avg_user_score, ROUND(AVG(r.critic_score), 1) AS avg_critic_score, COUNT(g.Name) AS num_games, ROUND(SUM(Total_Shipped)) AS total_sold
FROM game_sales_data AS g 
INNER JOIN reviews AS r USING (Name)
GROUP BY g.year
HAVING num_games > 50 AND avg_user_score >= 8.0 AND avg_critic_score >= 8.0
ORDER BY total_sold DESC
LIMIT 10

 * mysql+mysqldb://root:***@localhost/videogames
5 rows affected.


[(2010, 9.1, 8.4, 98, 607.0),
 (2011, 8.2, 8.6, 98, 507.0),
 (2007, 8.7, 8.0, 93, 426.0),
 (2009, 9.2, 8.6, 83, 414.0),
 (2008, 8.9, 8.1, 91, 406.0)]

<h1> Conclusion </h1>

This analysis helped us identify the best years for video games by examining four key metrics: average critic scores, average user scores, number of games sold per year, and total sales per year. Our findings revealed certain years that stood out for their critical acclaim, strong user reception, and commercial success in the video games industry. While individual metrics highlighted different standout years, including more fields and categories helped us find the years with the most impact in gaming history. This approach provided a better look at the industry's peak periods and offered insights into when the video game marked reached its highest points in both quality and popularity.